feat: generalized imports supporting pattern matching and selective field import (#3076)
dfinity/motoko: doc/modules/language-guide/pages/language-manual.adoc

  "canister:<name>"                 import external actor by <name>
```

> An import introduces a resource named '<pat>' referring to a local source module, module from a package of modules, or canister (imported as an actor).

Reviewer: ```suggestion
An import introduces a resource referring to a local source module, module from a package of modules, or canister (imported as an actor).
The contents of the resource are bound to `<pat>`. 

Though typically a simple identifier, `<id>`, `<pat>` can also be any composite pattern binding selective components of the resource.

The pattern must be irrefutable.
```

> The pattern is bound to a module and explicit bindings can be established

Reviewer: ```suggestion
```

Author: Somehow your eraser doesn't work :-(

> by using an object pattern.

Author: ```suggestion
```

  "canister:<name>"                 import external actor by <name>
```

An import introduces a resource referring to a local source module, module from a package of modules, or canister (imported as an actor).
The contents of the resource are bound to `<pat>`. 

Though typically a simple identifier, `<id>`, `<pat>` can also be any composite pattern binding selective components of the resource.

The pattern must be irrefutable.

> 
> by using an object pattern.

Author: ```suggestion
```

(Remark: when building multi-canister projects with the {sdk-short-name}, {proglang} programs can typically import canisters by alias (e.g. `import C "canister:counter"`), without specifying low-level canister ids (e.g. `import C "ic:lg264-qjkae"`).
The SDK tooling takes care of supplying the appropriate command-line arguments to the {proglang} compiler).

> (Remark: sensible choices for `<pat>` are identifiers, such as `Array`, or object patterns like `{ cons; nil = empty }`, which allow explicit importing of specified symbols only, including the possibility of changing the binding name).

Reviewer: ```suggestion
(Remark: sensible choices for `<pat>` are identifiers, such as `Array`, or object patterns like `{ cons; nil = empty }`, which allow selective importing of individual fields, under original or other names).
```

> (Remark: when building multi-canister projects with the {sdk-short-name}, {proglang} programs can typically import canisters by alias (e.g. `import C "canister:counter"`), without specifying low-level canister ids (e.g. `import C "ic:lg264-qjkae"`).

Reviewer: ```suggestion
(Remark: when building multi-canister projects with the {sdk-short-name}, {proglang} programs can typically import canisters by alias (e.g. ``import C "canister:counter"``), without specifying low-level canister ids (e.g. ``import C "ic:lg264-qjkae"``.)
```

Author: Let's change the remark above too?

Author: 3e8226ce041e941839e53d4803bdda4d8ce29298

The SDK tooling takes care of supplying the appropriate command-line arguments to the {proglang} compiler).

> (Remark: sensible choices for `<pat>` are identifiers, such as `Array`, or object patterns like `{ cons; nil = empty }`, which allow selective importing of individual fields, under original or other names).

Reviewer: ```suggestion
(Remark: sensible choices for `<pat>` are identifiers, such as ``Array``, or object patterns like ``{ cons; nil = empty }``, which allow selective importing of individual fields, under original or other names.)
```
